src/EasyAbp.FileManagement.EntityFrameworkCore/EasyAbp/FileManagement/Files/FileRepository.cs

```diff
@@ -120,11 +120,11 @@ public FileRepository(IDbContextProvider<IFileManagementDbContext> dbContextProv
             var nextNumber =
                 fileNames
                     .Select(x =>
-                        x.Substring(part1.Length, x.LastIndexOf(part2, StringComparison.Ordinal) - part1.Length))
+                        x.LastIndexOf(part2, StringComparison.Ordinal) > 0 ? x.Substring(part1.Length, x.LastIndexOf(part2, StringComparison.Ordinal) - part1.Length) : "")
```

Comment:
How about
```cshrap
                    .Where(x => x.EndsWith(part2, StringComparison.Ordinal))
```

Reply:
It's Ok，but why add new line？

Reply:
No need to select and parse unused items ("").

Reply:
If client upload a file end with ")",also need parse str,so I think both are fine,you decide.

Reply:
I prefer to keep the `.Select()` line unchanged and add
```csharp
                    .Where(x => x.EndsWith(part2, StringComparison.Ordinal))
```
for minimal impact.